### ISSUE TITLE
window.FirebasePlugin.grantPermission() IOS 10

### ISSUE THREAD

**Issue by eros2187** [arnesson/cordova-plugin-firebase#267]
Using the plugin I'm noticing that if I call the function as well as you see without calling the grantpermission () function the message is sent but I do not see any notification here is the code

`document.addEventListener('deviceready', onDeviceReady, false);

function onDeviceReady() {
$('#console').append('START');
setTimeout(getTheToken, 1000);
}

function getTheToken() {
window.FirebasePlugin.getToken(function(token) {
            if (token == null) {
                console.log("null token");
                setTimeout(getTheToken, 1000);
            } else {
                //window.FirebasePlugin.grantPermission();
                $('#console').append('token: ' + token);
                console.log("I got the token: " + token);
            }
        },
        function (error) {
            $('#consoleerr').append('error retrieving token: ' + error);
        }
    );
}
`
server
{"multicast_id":7649446427278374045,"success":1,"failure":0,"canonical_ids":0,"results":[{"message_id":"0:1494347311775201%812d9b5d812d9b5d"}]}

**If I lose the grantPermission () function, the server returns me**

`{"multicast_id":6314053970229096456,"success":0,"failure":1,"canonical_ids":0,"results":[{"error":"InvalidApnsCredential"}]}
my device is ipad mini 10.3.1 

**As can be solved?**

**Comment by jonnycross** [comment]
Also having the same issue on iOS10. 

window.FirebasePlugin.hasPermission always returns false even after window.FirebasePlugin.grantPermission. No errors returned on either.

Notification from the firebase console says success but window.FirebasePlugin.onNotificationOpen never gets fired.

Tested on iOS9 which works correctly so I know the server side, app and firebase console settings are all setup correctly.

Any ideas?

**Comment by viananicolas** [comment]
Same issue here. Anyone got it fixed?

**Comment by eros2187** [comment]
I have not solved what problems do you have?
Do you have the same problem with me?
I also expect someone so kind to give me some solution!!!!!!

**Comment by robertarnesson** [comment]
Looks like your apn certs are invalid (per the error message). see e.g. https://stackoverflow.com/questions/43515467/firebase-invalidapnscredential-while-certificates-are-valid

**Comment by jonnycross** [comment]
If the APN certs are invalid why would it still send to iOS 9 devices in production?

**Comment by eros2187** [comment]
The issue I posted does not concern the certificate because the certificate is trusted because it is still valid.
The problem rigurda instead of the plugin because as seen from my example I tested with ios 10 when I call the getTheToken () function WITHOUT RECOGNIZING THE FUNCTION THAT I HAD TO DISPLAY NOTES **window.FirebasePlugin.grantPermission ();** I do not receive any error on the certificate, in fact getting feedback on the server:
{"Multicast_id" 7649446427278374045, "success": 1, "failure": 0, "canonical_ids": 0, "results": [{"message_ID": "0: 1494347311775201% 812d9b5d812d9b5d"}]}
**But I do not see any notification on the device even if I'm successful!!**

If I ask permissions using the function
**window.FirebasePlugin.grantPermission ();**
  When I try to send the message from the server I get:

`{" Multicast_id "6314053970229096456," success ": 0," failure ": 1," canonical_ids ": 0," results ": [{" error ":" InvalidApnsCredential "}]}

I would like to understand why this happens? I used a ipad with version 10.3.
